matrixClass.js

```diff
@@ -0,0 +1,335 @@
+      endColumn = stepMatr.matrix[0].length;
+    }
+    for (let j = startColumn; j < endColumn; j++) {
+      const start = stepMatr.findValidRow(startRowInd, j);
+      if (start === -1) continue;
+      stepMatr.swapRow(start, startRowInd);
+      const startRow = stepMatr.matrix[startRowInd];
+      for (let i = startRowInd + 1; i < stepMatr.matrix.length; i++) {
+        const stepRow = stepMatr.matrix[i];
+        stepMatr.matrix[i] = Matrix.getRowForStepM(startRow, stepRow, j);
+      }
+      startRowInd++;
+    }
+    return stepMatr;
+  }
+
+  findValidRowFromBottom(startRowInd, startColInd) {
+    const nextInd = -1;
+    for (let i = startRowInd; i >= 0; i--) {
+      const elem = this.matrix[i][startColInd];
+      if (!elem) continue;
+      return i;
+    }
+    return nextInd;
+  }
+
+  getStepMatrReversed(startColumn, endColumn = 0) {
+    const stepMatr = Matrix.copy(this.matrix);
+    if (!(startColumn >= 0)) {
+      startColumn = stepMatr.matrix[0].length - 1;
+    }
+    let startRowInd = stepMatr.matrix.length - 1;
+    for (let j = startColumn; j >= endColumn; j--) {
+      const start = stepMatr.findValidRowFromBottom(startRowInd, j);
+      if (start === -1) continue;
+      stepMatr.swapRow(start, startRowInd);
+      const startRow = stepMatr.matrix[startRowInd];
+      for (let i = startRowInd - 1; i >= 0; i--) {
+        const stepRow = stepMatr.matrix[i];
+        stepMatr.matrix[i] = Matrix.getRowForStepM(startRow, stepRow, j);
+      }
+      startRowInd--;
+    }
+    return stepMatr;
+  }
+
+  static getSystemMatr(extendMatr) {
+    const matr = [];
+    const columnNum = extendMatr[0].length;
+    for (let i = 0; i < extendMatr.length; i++) {
+      matr[i] = extendMatr[i].slice(0, columnNum - 1);
+    }
+    return new Matrix(matr);
+  }
+
+  getRange() {
+    let nullRowNum = 0;
+    const stepMatr = this.getStepMatr();
+    const len = stepMatr.matrix.length;
+    for (let i = 0; i < len; i++) {
+      if (Matrix.isNullRow(stepMatr.matrix[i])) nullRowNum++;
+    }
+    return len - nullRowNum;
+  }
+
+  //SoLE = System of Linear Equations
+
+  static systemHasntSol(extendMatr) {
+    const systemMatr = Matrix.getSystemMatr(extendMatr.matrix);
+    const sysRang = systemMatr.getRange();
+    const extRang = extendMatr.getRange();
+    const varNum = systemMatr.matrix[0].length;
+    if (extRang > sysRang) {
+      return 'System have not any solutions';
+    }
+    if (extRang !== varNum) {
+      return 'System have infinitely many solutions';
+    }
+    return false;
+  }
+
+  static findSolOfSoLE(extendMatr) {
+    const columnIndex = extendMatr.matrix[0].length - 2;
+    const systHasntSol = Matrix.systemHasntSol(extendMatr);
+    if (systHasntSol) {
+      return systHasntSol;
+    }
+    const step1Matr = extendMatr.getStepMatr(columnIndex);
+    const step2Matr = step1Matr.getStepMatrReversed(columnIndex);
+    const columnNum = step2Matr.matrix[0].length;
+    let res = 'Solution:\n';
```

Comment:
The solver should not concern itself with the way results are presented, this should just return the results and not a string.